pkg/controller/metadata/metadata_controller.go

```diff
@@ -279,8 +281,18 @@ func (r *ReconcileMetadata) Reconcile(request reconcile.Request) (reconcile.Resu
 		}
 	} else if err != nil {
 		return reconcile.Result{}, err
-	} else {
-		// TODO: we may want to handle updates to self-heal metadata, but this would need to be more inteligent than below
+	} else if foundSecret.ObjectMeta.Annotations[versionAnnotation] != currentVersion {
+		log.Info("Updating Secret", "namespace", instance.Namespace, "name", instance.Name)
+		updatedData, err := r.generateMetadataSecretData(instance, hsmCreds, hsmCreds)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		foundSecret.ObjectMeta.Annotations[versionAnnotation] = currentVersion
+		foundSecret.Data = updatedData
+		err = r.Update(context.TODO(), foundSecret)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to update Secret %s: %s", foundSecret.ObjectMeta.Name, err)
+		}
```

Comment:
Could an `info` log message be added to indicate successful completion?

Reply:
Sure ... Added a commit to pair up `creating-x` with `created-x` and `updating-x` with `updated-x` messages